std/assembly/dataview.ts

```diff
@@ -1,49 +1,57 @@
-import {
-  loadUnsafeWithOffset,
-} from "./internal/arraybuffer";
+import { HEADER_SIZE } from "./internal/arraybuffer";
+
+@inline
+function get<T> (buffer: ArrayBuffer, byteOffset: i32, littleEndian: boolean): T {
+  const result = load<T>(changetype<usize>(buffer) + byteOffset, HEADER_SIZE);
+  return littleEndian ? result : bswap<T>(result);
+}
 
 export class DataView {
-  readonly buffer: ArrayBuffer;
-  readonly byteOffset: i32;
-  readonly byteLength: i32;
-
-  constructor (buffer: ArrayBuffer, byteOffset: i32, byteLength: i32) {
-    this.buffer = buffer
-    this.byteOffset = byteOffset
-    this.byteLength = byteLength
-  }
-
-  @inline
-  getUint32 (byteOffset: u32, littleEndian: boolean): u32 {
-    if (littleEndian) {
-      return (
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 0, this.byteOffset) << 24) &
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 1, this.byteOffset) << 16) &
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 2, this.byteOffset) << 8) &
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 3, this.byteOffset) << 0)
-      );
-    } else {
-      return (
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 0, this.byteOffset) << 0) &
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 1, this.byteOffset) << 8) &
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 2, this.byteOffset) << 16) &
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 3, this.byteOffset) << 24)
-      );
-    }
-  }
-
-  @inline
-  getUint16 (byteOffset: u32, littleEndian: boolean): u16 {
-    if (littleEndian) {
-      return (
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 0, this.byteOffset) << 8) &
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 1, this.byteOffset) << 0)
-      );
-    } else {
-      return (
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 0, this.byteOffset) << 0) &
-        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 1, this.byteOffset) << 8)
-      );
-    }
+  constructor(
+    readonly buffer: ArrayBuffer,
+    readonly byteOffset: i32 = 0,
+    readonly byteLength: i32 = i32.MAX_VALUE,
+  ) {
+    this.byteLength = min(buffer.byteLength, byteLength);
```

Comment:
Should also take care of negative values I think. What happens n JS if a negative value is provided?

Reply:
```ts
if (byteOffset < 0 || byteOffset > byteLength) {
  throw new Error('Start offset is outside the bounds of the buffer');
}
if (byteLength < 0) throw new Error('Invalid DataView length');

this.byteLength = min(buffer.byteLength, byteLength);
```

Reply:
I've added checks that works exactly as in the browser 👌 (5358cff)